src/app/manageinvoices/manageinvoices.component.ts

```diff
@@ -7,63 +13,102 @@ import { Component, OnInit } from '@angular/core';
-  {"id":6,"CustomerName":"John Doe","SupplierID":3,"Date":"12/6/2021",DueDate:"12/6/2022","GasAmount":763,ElectricityAmount:0,"Price":167,"Tax":5.70, ClientType: "Business", StatusID: 1, Status: "Generated"}];
+  invoices! : Invoice[];
+  DetailsClicked : boolean = false;
+  DetailsID : number = 0;
+  LogonError : boolean = false;
+  UserMan : boolean = false;
 
-  constructor() {}
+  constructor(private invoiceService : InvoiceService, public auth: AuthService) {}
+  AllViewer: UserRole[] = [UserRole.ACCOUNTANT,UserRole.ADMIN, UserRole.EMPLOYEE, UserRole.HR_MANAGER, UserRole.MANAGER, UserRole.SUPPORT, UserRole.TECHNICIAN];
 
   ngOnInit(): void {
+    let MyRole = this.auth.getUserRoleId();
+    if(MyRole?.some(r => this.AllViewer.includes(r))!)
+    {
+      this.invoiceService.GetAllInvoices().subscribe(invoices => {this.invoices = invoices; console.log(this.invoices);});
+    }
+    else if (MyRole?.includes(UserRole.CUSTOMER))
+    {
+      this.invoiceService.GetUserInvoices().subscribe(invoices => {this.invoices = invoices; console.log(this.invoices);});
+      this.UserMan = true;
+    }    
+    else
+    {
+      this.LogonError = true;
+    }
+  }
+ 
+  dateFormat(date : Date) : string
+  {
+    return formatDate(date, 'MM/dd/yyyy', 'en');
   }
 
-  public open(id: number): void {
-    let invoice = new Invoice();
-    this.invoices.forEach(function (inv) {
-      if (inv.id == id) {
-        invoice = new Invoice(inv.id, inv.CustomerName, inv.SupplierID.toString(), inv.Date, inv.DueDate, inv.GasAmount, inv.ElectricityAmount, inv.Price, inv.Tax, "", inv.StatusID, inv.Status);
-      }
-    });
-    this.invoicedetails = invoice;
-    return;
+  GetCustomerType(customer : CustomerType) : string
+  {
+    return CustomerType[Number(customer)];
+  }
+
+  GetInvoiceType(customer : INVOICE_TYPE) : string
+  {
+    return INVOICE_TYPE[Number(customer)];
+  }
+  
+  GetInvoiceStatus(customer : INVOICE_STATUS) : string
+  {
+    return INVOICE_STATUS[Number(customer)];
+  }
+
+  CheckInvoiceStatus(customer : INVOICE_STATUS) : boolean
+  {
+    if(customer == INVOICE_STATUS.PAID)
+    {
+      return false;
+    }
+    return true;
   }
-}
 
-export class Invoice {
-  public id: number;
-  public customerName: string;
-  public supplierName: string;
-  public date: string;
-  public dueDate: string;
-  public gasAmount: number;
-  public electricityAmount: number;
-  public price: number;
-  public tax: number;
-  public contractType: string;
-  public statusID: number;
-  public status: string;
+  OnClickDetails(i : number)
+  {
+    this.DetailsID = i;
+    this.DetailsClicked = true;
+    this.DownloadPDF()
+  }
+
+  CheckUserMan(i : number | undefined) : boolean
+  {
+    console.log(i);
+    if(i != this.auth.getUserId())
+    {
+      return false;
+    }
+    return true;
+  }
```

Comment:
also role/auth guards

Reply:
this too had a reason I can't remember fully